src/services/firebase.test.ts

```diff
@@ -72,3 +72,44 @@ it('adds a new match in firebase', () => {
   };
   ourFirebase.createMatch(gameInfo);
 });
+
+it('fetch match list from firebase', done => {
+  const matchInfo: fbr.Match = {
+    participants: {},
+    createdOn: 123456,
+    lastUpdatedOn: 123456,
+    gameSpecId: 'Test Game Spec',
+    pieces: {}
+  };
+
+  firebase
+    .database()
+    .ref('/gamePortal/matches')
+    .child('Test Match Id')
+    .set(matchInfo);
+
+  ourFirebase.listenToMyMatchesList(addMatch);
+
+  function addMatch() {
+    const currentUser = firebase.auth().currentUser;
+    if (!currentUser) {
+      throw new Error('You must be logged in');
+    }
+    const userId = currentUser.uid;
+    firebase
```

Comment:
Assert something, what are you testing here.

Reply:
Discussed with professor, you don't need to assert anything. Just make sure you push all of your promises in `ourFirebase.allPromisesForTests` array.

Reply:
You must use the method 
  refSet(ref, value);

instead of doing:
  ref.set(value)



refSet will check that you don't get errors, and the test will wait for the promise to resolve.

Reply:
But refSet function is in firebase.ts, and is marked as non-exported functions, should I write another refSet in firebase.test.ts?

Reply:
you can export it for now :)

I'll later try to fix the test code to use methods in ourFirebase (e.g. to create a match).